languageFactories/wdl-draft2/src/main/scala/languages/wdl/draft2/WdlDraft2LanguageFactory.scala

```diff
@@ -54,19 +58,23 @@ class WdlDraft2LanguageFactory(override val config: Map[String, Any]) extends La
       list.sequence[Checked, Unit].void
     }
 
-    import common.validation.Validation._
-
-    lazy val wdlNamespaceValidation: ErrorOr[WdlNamespaceWithWorkflow] = source match {
-      case w: WorkflowSourceFilesWithDependenciesZip =>
-        for {
-          importsDir <- LanguageFactoryUtil.validateImportsDirectory(w.importsZip)
-          wf <- WdlNamespaceWithWorkflow.load(workflowSource, importResolvers map resolverConverter).toErrorOr
-          _ = importsDir.delete(swallowIOExceptions = true)
-        } yield wf
-      case _: WorkflowSourceFilesWithoutImports =>
-        WdlNamespaceWithWorkflow.load(workflowSource, importResolvers map resolverConverter).toErrorOr
+    def workflowHashKey: String = {
+      workflowSource.md5Sum + (source.importsZipFileOption map { bytes => new String(MessageDigest.getInstance("MD5").digest(bytes)) }).getOrElse("")
```

Comment:
I think so but to double check - are we happy to assert that MD5suming the zip file is less CPU intensive than just parsing the WDL?

Reply:
It seems a pretty safe assumption that digesting the array of bytes is cheaper than stringifying it and parsing. 🙂